src/components/MatrixBlock/index.js

```diff
@@ -0,0 +1,88 @@
+import { nanoid } from "nanoid";
+import { CustomButton } from "../shared/CustomButton";
+import styles from "./MatrixBlock.module.scss";
+import CancelIcon from "@mui/icons-material/Cancel";
+import { TableCell } from "../MatrixTableCell";
+
+export const Matrix = ({ data, settings, setSettings, handleSetMatrix }) => {
+  const handleAddRow = () => {
+    handleSetMatrix([
+      ...data,
+      Array(settings.columns)
+        .fill()
+        .map(() => ({
+          id: nanoid(),
+          num: Math.floor(Math.random() * (999 - 100) + 100),
+        })),
+    ]);
+    setSettings({ ...settings, rows: data.length + 1 });
+  };
+
+  const getAverageOfSum = () => {
+    let avgSum = 0;
+    data.map((item) => {
+      avgSum += item.reduce((acc, curr) => (acc += curr.num), 0);
+    });
+    return Math.round(avgSum / settings.rows);
+  };
+
+  return data.length === 0 ? null : (
+    <div className="app__matrix-block">
+      <CustomButton data="Add row" onClick={handleAddRow} />
+      <table className={styles.app__matrixTable}>
+        <thead>
+          <tr>
+            <TableCell data="№" />
+            {data[0].map((item, index) => (
```

Comment:
let's move it own const headings that will be result of function getHeadings = (columns) => Array(columns).fill(0).map(_,index => index+1)

Reply:
and here headings.map(item => <TableCell data={item} />